### ISSUE TITLE
Nodes as legends example is not aligning legend

### ISSUE THREAD

**Issue by juanpastas** [almende/vis#1857]
![nodes not aligned](https://cloud.githubusercontent.com/assets/671721/15301472/5f45b142-1b73-11e6-8cc8-3ed7fdc8d1b5.png)

Whenever I reload I get different positions. Do you know why is this caused?

Example is [here](http://visjs.org/examples/network/exampleApplications/nodeLegend.html).

**Comment by Stexxen** [comment]
Just coming here to create the same issue. The example at http://visjs.org/examples/network/exampleApplications/nodeLegend.html 
shows the legend incorrectly.
As of 7th Jun, the example is using version 4.16.1

The code is setting the x to a fixed value as per here
```
nodes.push({id: 1000, x: x, y: y, label: 'Internet', group: 'internet', value: 1, fixed: true, physics:false});
nodes.push({id: 1001, x: x, y: y + step, label: 'Switch', group: 'switch', value: 1, fixed: true,  physics:false});
nodes.push({id: 1002, x: x, y: y + 2 * step, label: 'Server', group: 'server', value: 1, fixed: true,  physics:false});
nodes.push({id: 1003, x: x, y: y + 3 * step, label: 'Computer', group: 'desktop', value: 1, fixed: true,  physics:false});
nodes.push({id: 1004, x: x, y: y + 4 * step, label: 'Smartphone', group: 'mobile', value: 1, fixed: true,  physics:false});
```
So x and y are being set correctly, but there are clearly in the wrong place. Testing across browsers. I have the following

Chrome  50.0.2661.102 on OSX 10.11.5
![image](https://cloud.githubusercontent.com/assets/4494301/15869731/938e0124-2ce5-11e6-8a74-444bb6d48381.png)

Safari 9.1.1 on OSX 10.11.5
![image](https://cloud.githubusercontent.com/assets/4494301/15869776/cad3adaa-2ce5-11e6-89f4-c2db102b0530.png)

Edge 38.14342.1000.0 on Windows 10 build 14342
![image](https://cloud.githubusercontent.com/assets/4494301/15869798/e26d015a-2ce5-11e6-9389-06c9aeca810b.png)

**Comment by Stexxen** [comment]
looks like a bug with the layout.improvedLayout

`layout.improvedLayout : true`
![image](https://cloud.githubusercontent.com/assets/4494301/17083269/28353862-5192-11e6-87bb-c218dfbe7525.png)

`layout.improvedLayout : false`
![image](https://cloud.githubusercontent.com/assets/4494301/17083271/466b9fba-5192-11e6-9fcb-3be4c89537b5.png)


I should also add with `layout.improvedLayout : true`, it only seems breaks once the total nodes goes above  10..

**Comment by mojoaxel** [comment]
fixed with eb675cd7f64da1e5bdbf2cd3b151d7c3df3e12c3
